lib/hyrax/transactions/container.rb

```diff
@@ -18,23 +18,57 @@ module Transactions
     #
     # @see https://dry-rb.org/gems/dry-container/
     class Container
+      require 'hyrax/transactions/apply_change_set'
       require 'hyrax/transactions/create_work'
       require 'hyrax/transactions/destroy_work'
+      require 'hyrax/transactions/work_create'
       require 'hyrax/transactions/steps/apply_collection_permission_template'
       require 'hyrax/transactions/steps/apply_permission_template'
       require 'hyrax/transactions/steps/apply_visibility'
       require 'hyrax/transactions/steps/destroy_work'
       require 'hyrax/transactions/steps/ensure_admin_set'
       require 'hyrax/transactions/steps/ensure_permission_template'
+      require 'hyrax/transactions/steps/save'
       require 'hyrax/transactions/steps/save_work'
       require 'hyrax/transactions/steps/set_default_admin_set'
       require 'hyrax/transactions/steps/set_modified_date'
-      require 'hyrax/transactions/steps/set_uploaded_date'
+      require 'hyrax/transactions/steps/set_uploaded_date_unless_present'
+      require 'hyrax/transactions/steps/validate'
 
       extend Dry::Container::Mixin
 
       # Disable BlockLength rule for DSL code
       # rubocop:disable Metrics/BlockLength
+      namespace 'change_set' do |ops|
+        ops.register 'apply' do
+          ApplyChangeSet.new
+        end
+
+        ops.register 'ensure_admin_set' do
+          Steps::EnsureAdminSet.new
+        end
+
+        ops.register 'save' do
+          Steps::Save.new
+        end
+
+        ops.register 'set_default_admin_set' do
+          Steps::SetDefaultAdminSet.new
+        end
+
+        ops.register 'set_modified_date' do
+          Steps::SetModifiedDate.new
+        end
+
+        ops.register 'set_uploaded_date_unless_present' do
```

Comment:
One of the most liberating moments in software development naming is getting to the point where the classes are descriptive and verbose (and even include verbs). I found as I moved towards more descriptive classes, the full breadth of language (and thus symbols) became available. I love this class name!